inlong-manager/manager-workflow/src/main/java/org/apache/inlong/manager/workflow/definition/ServiceTask.java

```diff
@@ -42,6 +44,8 @@
 
     private ServiceTaskType serviceTaskType;
 
+    private boolean isInit;
```

Comment:
For isInit

If there is multi-threaded access, it is recommended to add volatile define. If there is a multi-threaded concurrent modification, at least a lock is required to avoid reentrancy

Reply:
Excellent comment !!

Reply:
I prefer to use CAS to avoid multi-thread modification like this;

private AtomicBoolean isInit = new AtomicBoolean(false);

public void initListeners(WorkflowContext workflowContext) {
        if (isInit.compareAndSet(false, true)) {
            if (listenerProvider == null || serviceTaskType == null) {
                return;
            }
            Iterable<TaskEventListener> listeners = listenerProvider.get(workflowContext, serviceTaskType);
            addListeners(Lists.newArrayList(listeners));
        } else {
            log.debug("ServiceTask:{} is already init", getName());
        }
    }